Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -573,6 +578,110 @@ namespace AzToolsFramework
+        {
+            using namespace AzFramework::AssetSystem;
+            EntryTypeFilter* foldersFilter = new EntryTypeFilter();
+            foldersFilter->SetEntryType(AssetBrowserEntry::AssetEntryType::Folder);
+
+            auto selection = AzToolsFramework::AssetBrowser::AssetSelectionModel::EverythingSelection();
+            selection.SetTitle("folder to move to");
+            selection.SetMultiselect(false);
+            selection.SetDisplayFilter(FilterConstType(foldersFilter));
+            AssetBrowserTreeViewDialog dialog(selection, this);
+            dialog.exec();
+
+            const AZStd::vector<AZStd::string> folderPaths = selection.GetSelectedFilePaths();
+
+            if (!folderPaths.empty())
+            {
+                AZStd::string folderPath = folderPaths[0];
+                AZ_TracePrintf("JJS", "FolderPath = %s\n", folderPath.c_str());
+                bool connectedToAssetProcessor = false;
+                AzFramework::AssetSystemRequestBus::BroadcastResult(
+                    connectedToAssetProcessor, &AzFramework::AssetSystemRequestBus::Events::AssetProcessorIsReady);
+
+                if (connectedToAssetProcessor)
+                {
+                    auto entries = GetSelectedAssets();
+
+                    for (auto entry : entries)
+                    {
+                        using namespace AZ::IO;
+                        AZStd::string originalFname;
+                        AssetBrowserEntry* item = entry;
+                        Path fromPath = item->GetFullPath();
+                        PathView filename = fromPath.Filename();
+                        Path toPath(folderPath);
+                        toPath /= filename;
+                        AssetChangeReportRequest request(
+                            AZ::OSString(fromPath.c_str()), AZ::OSString(toPath.c_str()), AssetChangeReportRequest::ChangeType::CheckMove);
+                        AssetChangeReportResponse response;
+
+                        if (SendRequest(request, response))
+                        {
+                            AZStd::string message;
+                            for (int i = 0; i < response.m_lines.size(); ++i)
+                            {
+                                message += response.m_lines[i] + "\n";
+                            }
+
+                            if (message.size())
+                            {
+                                QMessageBox msgBox(this);
+                                msgBox.setWindowTitle("Before Move Asset Information");
+                                msgBox.setIcon(QMessageBox::Warning);
+                                msgBox.setText("The asset you are moving may be referenced in other assets.");
+                                msgBox.setInformativeText("More information can be found by pressing \"Show Details...\".");
+                                QAbstractButton* moveButton = (QAbstractButton*)msgBox.addButton("Move", QMessageBox::YesRole);
```

Comment:
Shouldn't need this `(QAbstractButton*)` cast

Reply:
It doesn't compile without it.
: error C2440: 'initializing': cannot convert from 'QPushButton *' to 'QAbstractButton *'
: error C2440:                                 QAbstractButton* moveButton = msgBox.addButton("Move", QMessageBox::YesRole);
: error C2440:                                                                                                             ^
 message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
 message :                                 QAbstractButton* moveButton = msgBox.addButton("Move", QMessageBox::YesRole);
 message :                                                             ^

Reply:
@alexmontAmazon  Is correct here, if you set the type as `auto*` or even explicitly to `QPushButton*` the comparison below to `msgBox.clickedButton()` will compile (there are several instances of this in the code-base).

Reply:
I still need
                                if (msgBox.clickedButton() == reinterpret_cast<QAbstractButton *>(moveButton))
error C2446: '==': no conversion from 'QPushButton *' to 'QAbstractButton *'
error C2446:                                 if (msgBox.clickedButton() == moveButton)
error C2446:                                                                         ^
message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
message :                                 if (msgBox.clickedButton() == moveButton)
message :                                                               ^